src/verify.rs

```diff
@@ -5,25 +5,52 @@ use policy_fetcher::verify::Verifier;
 use std::io::prelude::*;
 use std::{collections::HashMap, fs::File};
 
+pub(crate) type VerificationAnnotations = HashMap<String, String>;
+
 pub(crate) async fn verify(
     url: &str,
-    docker_config: Option<DockerConfig>,
-    sources: Option<Sources>,
-    annotations: Option<HashMap<String, String>>,
+    docker_config: &Option<DockerConfig>,
+    sources: &Option<Sources>,
+    annotations: &Option<VerificationAnnotations>,
```

Comment:
Move the reference inside of the `Option`. You can then turn something like `Option<DockerConfig>` into `Option<&DockerConfig>` by doing:

```rust
let docker_config: Option<DockerConfig> = Some(DockerConfig::default());
let dc = Option<&DockerConfig> = docker_config.as_ref();
```

Reply:
Yes, I would tackle that step by step. Maybe once we're done with this change and my work on the "get policy manifest" callback we can make a single PR that changes all of that.

Does it make sense?

Reply:
Yes, thanks for the clarification!

Done in https://github.com/kubewarden/kwctl/pull/84/commits/ebee37c89eb96ab442e2b13c5267829e3fdbd93e.